drivers/gpio/gpio_mcux_rgpio.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2023, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_imx_gpio
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/irq.h>
+#include <fsl_common.h>
+#include <fsl_rgpio.h>
+
+#ifdef CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+#endif
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+struct gpio_pin_gaps {
+	uint8_t start;
+	uint8_t len;
+};
+
+/* Required by DEVICE_MMIO_NAMED_* macros */
+#define DEV_CFG(_dev) \
+	((const struct mcux_rgpio_config *)(_dev)->config)
+#define DEV_DATA(_dev) ((struct mcux_rgpio_data *)(_dev)->data)
+
+struct mcux_rgpio_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+
+	DEVICE_MMIO_NAMED_ROM(reg_base);
+
+#ifdef CONFIG_PINCTRL
+	const struct pinctrl_soc_pinmux *pin_muxes;
+	const struct gpio_pin_gaps *pin_gaps;
+	uint8_t mux_count;
+	uint8_t gap_count;
+#endif
+};
+
+struct mcux_rgpio_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data general;
+
+	DEVICE_MMIO_NAMED_RAM(reg_base);
+
+	/* port ISR callback routine address */
+	sys_slist_t callbacks;
+};
+
+static int mcux_rgpio_configure(const struct device *dev,
+				gpio_pin_t pin, gpio_flags_t flags)
+{
+	RGPIO_Type *base = (RGPIO_Type *)DEVICE_MMIO_NAMED_GET(dev, reg_base);
+	const struct mcux_rgpio_config *config = dev->config;
+
+#ifdef CONFIG_PINCTRL
```

Comment:
I'm not sure we need the `CONFIG_PINCTRL` dependencies anymore. These were generally removed, see #57042

Reply:
removed.